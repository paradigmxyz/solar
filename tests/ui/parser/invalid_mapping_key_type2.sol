contract B {
    uint x;
    uint y;
}
contract C {
    mapping(B.x => uint) m0; // Identifier path as mapping key type is allowed
    mapping(B.x => mapping(B.x => B.y)) m1;
    mapping([] => uint) m2; //~ ERROR: expected one of elementary type name or path, found `[`
}


