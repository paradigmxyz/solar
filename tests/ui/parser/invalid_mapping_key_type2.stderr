error: expected one of elementary type name or path, found `[`
  --> ROOT/tests/ui/parser/invalid_mapping_key_type2.sol:LL:CC
   |
LL |     mapping([] => uint) m2;
   |             ^ expected one of elementary type name or path

error: aborting due to 1 previous error

